app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentKotlinActivity.kt

```diff
@@ -19,32 +22,35 @@ import com.getsensibill.web.ui.WebUiNetworkErrorFragment
  * Also, please use, or see [WebUiNetworkErrorFragment] as reference for what to display when
  * [Listener.onDisplayNetworkError] is called.
  */
-class FragmentKotlinActivity : AppCompatActivity(), WebUiFragment.Listener {
+class FragmentKotlinActivity : AppCompatActivity(),
+    WebUiFragment.Listener, WebUiNetworkErrorFragment.Listener {
+
     private lateinit var binding: ActivityFragmentKotlinBinding
 
+    @IdRes
+    private val webContainerId: Int = R.id.fragment_container
+
+    /** The instance of [WebUiFragment] that is being displayed by the activity */
+    private val webFragment: WebUiFragment?
+        get() = supportFragmentManager.findFragmentByTag(TAG_WEB_FRAGMENT) as? WebUiFragment
+
+    private val errorFragment: WebUiNetworkErrorFragment?
+        get() = supportFragmentManager.findFragmentByTag(TAG_WEB_NETWORK_ERROR_FRAGMENT) as? WebUiNetworkErrorFragment
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityFragmentKotlinBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        if(savedInstanceState == null) {
-            val fragment = WebUiFragment().apply {
-                arguments = Bundle().apply {
-                    // Pass in a navigation override. Defaults as .DASHBOARD
-                    putParcelable(WebUiFragment.ARG_NAVIGATION_OVERRIDE, NavigationIntent.DASHBOARD)
-                    // Pass in a custom Theme override. Defaults to null
-                    putParcelable(
-                        WebUiFragment.ARG_WEB_THEME_OVERRIDE,
-                        WebTheme(Brand())
-                    )
-                }
-            }
-            supportFragmentManager.commit {
-                replace(binding.fragmentContainer.id, fragment, "FRAGMENT_TAG")
-            }
+        if (savedInstanceState == null) {
+            loadWebUi()
         }
     }
 
+    override fun onBackPressed() {
```

Comment:
you've added `onBackPressed` here, but not to the java version - should it be included there as well?

Reply:
i think it's unnecessary in either example. removed